haproxy_stats.sh

```diff
@@ -61,28 +100,26 @@ fi
 #   check if HAPROXY socket is writable
 # if we are NOT getting stats:
 #   check if we can read the stats cache file
-if [ "$GET_STATS" -eq 1 ]
-then
-  if [ -e "$CACHE_FILEPATH" ] && [ ! -w "$CACHE_FILEPATH" ]
-  then
-    echo 'ERROR: stats cache file exists, but is not writable'
-    exit 126
-  elif [ ! -w ${CACHE_FILEPATH%/*} ]
-  then
-    echo 'ERROR: stats cache file path is not writable'
-    exit 126
-  fi
-  if [[ $QUERYING_METHOD == "SOCKET" && ! -w $HAPROXY_SOCKET ]]
-  then
-    echo "ERROR: haproxy socket is not writable"
-    exit 126
-  fi
-elif [ ! -r "$CACHE_FILEPATH" ]
-then
-  echo 'ERROR: cannot read stats cache file'
-  exit 126
+if [ "$GET_STATS" -eq 1 ]; then
+    if [ -e "$CACHE_STATS_FILEPATH" ]; then
+        if [ ! -w "$CACHE_STATS_FILEPATH" ]; then
+            fail 126 'ERROR: stats cache file exists, but is not writable'
+        elif [ ! -s "$CACHE_STATS_FILEPATH" ]; then
+            debug "ERROR: stats cache file exists, but it's empty -> destroying it!"
+            rm -f "$CACHE_STATS_FILEPATH"
+            if [ $? -ne 0 ]; then
+                fail 126 "ERROR: problems deleting cache file, please check permissions!"
+            fi
+        fi
+    fi
+    if [[ $QUERYING_METHOD == "SOCKET" && ! -w $HAPROXY_SOCKET ]]; then
+        fail 126 "ERROR: haproxy socket is not writable"
+    fi
+elif [ ! -r "$CACHE_STATS_FILEPATH" ]; then
+    fail 126 'ERROR: cannot read stats cache file'
```

Comment:
why not put the fail condition into else or the "if" you've started in line 104?

Reply:
To which `fail` are you referring to?

I didn't changed the logic of that nested `if` I've just converted the `echo`&`exit` into `fail` and added the caching file removal in case the file exists but it's empty (that happened to me a lot while using the original implementation).